core/src/loader.rs

```diff
@@ -1604,38 +1609,116 @@ impl<'gc> Loader<'gc> {
     }
 
     /// Report a movie loader completion to script code.
-    fn movie_loader_complete(handle: Index, uc: &mut UpdateContext<'_, 'gc>) -> Result<(), Error> {
-        let (clip, event_handler) = match uc.load_manager.get_loader_mut(handle) {
+    fn movie_loader_complete(
+        handle: Index,
+        uc: &mut UpdateContext<'_, 'gc>,
+        dobj: Option<DisplayObject<'gc>>,
+    ) -> Result<(), Error> {
+        let (target_clip, event_handler, movie) = match uc.load_manager.get_loader_mut(handle) {
             Some(Loader::Movie {
                 target_clip,
+                movie,
                 event_handler,
                 ..
-            }) => (*target_clip, *event_handler),
+            }) => (*target_clip, *event_handler, movie.clone()),
             None => return Err(Error::Cancelled),
             _ => unreachable!(),
         };
 
+        let loader_info =
+            if let Some(MovieLoaderEventHandler::Avm2LoaderInfo(loader_info)) = event_handler {
+                Some(*loader_info.as_loader_info_object().unwrap())
+            } else {
+                None
+            };
+
+        if let Some(loader_info) = loader_info {
+            // Store the real movie into the `LoaderStream`, so that
+            // 'bytesTotal' starts returning the correct value
+            // (we previously had a fake empty SwfMovie).
+            // However, we still use `LoaderStream::NotYetLoaded`, since
+            // the actual MovieClip display object has not run its first
+            // frame yet.
+            loader_info.set_loader_stream(
+                LoaderStream::NotYetLoaded(movie.clone().unwrap(), Some(dobj.unwrap()), false),
+                uc.gc_context,
+            );
+        }
+
+        if let Some(dobj) = dobj {
+            if dobj.as_movie_clip().is_some() {
+                // We call these methods after we initialize the `LoaderInfo`, but before the
+                // add the the loaded clip as a child. The frame constructor should see
```

Comment:
(Nit: It would be nice if the wording in these comments from the previous version were fixed, or the comments are removed).

Nit: Are the comments still accurate?

Reply:
Yes, this comment is still accurate.

Reply:
(Nit: Even though this was written before your PR, I would really like it if the wording in this part of the comment: `but before the add the the loaded` was changed).